presto-spi/src/main/java/com/facebook/presto/spi/SchemaUtil.java

```diff
@@ -25,7 +27,7 @@ static String checkNotEmpty(String value, String name)
             throw new NullPointerException(name + " is null");
         }
         if (value.isEmpty()) {
-            throw new IllegalArgumentException(name + " is empty");
+            throw new PrestoException(GENERIC_USER_ERROR, name + " is empty");
```

Comment:
This is not the right place so handle this, as this method does not even know what `name` is supposed to represent. This is simply `requireNonNull + checkArgument`.
IMO you should do the check in the proper place, where you handle user input.

Reply:
Will do, thanks @findepi !

Reply:
Question: should this be a parsing error similar to when the catalog/schema/table/column name is null?
For now assuming yes.